src/components/PostDetails.tsx

```diff
@@ -1,117 +1,143 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  selectedPost: Post | null,
+  isCommentButtonClicked: boolean,
+  setIsCommentButtonClicked: (value: boolean) => void,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  isCommentButtonClicked,
+  setIsCommentButtonClicked,
+}) => {
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [commentErrorMessage, setCommentErrorMessage] = useState('');
+  const [postsComments, setPostsComments] = useState<Comment[]>([]);
+
+  const commentButtonHandler = () => {
+    setIsCommentButtonClicked(true);
+  };
+
+  const getPostsComments = () => {
+    return client.get<Comment[]>(`/comments?postId=${selectedPost?.id}`)
+      .catch(() => {
+        setCommentErrorMessage('Something went wrong!');
+      });
+  };
+
+  useEffect(() => {
+    setIsCommentsLoading(true);
+    getPostsComments()
+      .then(comentsFromServer => {
+        if (comentsFromServer) {
+          setPostsComments(comentsFromServer);
+        }
+      })
+      .finally(() => setIsCommentsLoading(false));
+  }, [selectedPost]);
+
+  const removeComment = (id: number) => {
+    return client.delete(`/comments/${id}`);
+  };
+
+  const deleteCommentHandler = (id: number) => {
+    setIsCommentsLoading(true);
+    setPostsComments(prew => prew.filter(CurentComment => (
+      id !== CurentComment.id
+    )));
```

Comment:
callback parameter name can't be Capitalize 
```suggestion
    setPostsComments(prew => prew.filter(curentComment => (
      id !== curentComment.id
    )));
```